scripts/performance/llm/finetune_llama4_e128.py

```diff
@@ -0,0 +1,264 @@
+    set_primary_perf_configs,
+    slurm_executor,
+)
+
+HF_MODEL_URI = "meta-llama/Llama-4-Maverick-17B-128E-Instruct"
+
+
+def override_recipe_configs(
+    args: str,
+    num_nodes: int,
+    mbs: int,
+    gbs: int,
+    tp_size: int,
+    pp_size: int,
+    cp_size: int,
+    vp_size: int,
+    ep_size: int,
+    num_layers: int,
+    hidden_size: int,
+    etp_size: int,
+    enable_cuda_graphs: bool,
+    use_mcore_fsdp: bool,
+    recompute_layers: int,
+    activation_offload_layers: int,
+):
+    """
+    Llama4 e128 fine-tuning recipe aimed at achieving best possible performance.
+    Only supports SFT (Supervised Fine-Tuning).
+
+    NOTE: Use fp8 precision training with caution. It might not give desirable results.
+    """
+    assert args.finetuning == "sft", "Only SFT (Supervised Fine-Tuning) is supported"
+    finetuning_scheme = "none" if args.finetuning == "sft" else args.finetuning
+
+    recipe = finetune_recipe(peft_scheme=finetuning_scheme, performance_mode=True, packed_sequence=True)
+
+    recipe = set_primary_perf_configs(
+        recipe,
+        finetuning_scheme,
+        num_nodes,
+        args.gpus_per_node,
+        mbs,
+        gbs,
+        args.max_steps,
+        tp_size,
+        pp_size,
+        cp_size,
+        vp_size,
+        ep_size,
+        num_layers,
+        hidden_size,
+        etp_size,
+        enable_cuda_graphs=enable_cuda_graphs,
+        use_mcore_fsdp=use_mcore_fsdp,
+        recompute_layers=recompute_layers,
+        activation_offload_layers=activation_offload_layers,
+        compute_dtype=args.compute_dtype,
+        fp8_recipe=args.fp8_recipe,
+    )
+
+    recipe = set_exp_logging_configs(
+        recipe,
+        finetuning_scheme,
+        "llm",
+        "llama4",
+        args.tensorboard,
+        args.wandb,
+        args.wandb_prj_name,
+        args.wandb_job_name,
+    )
+
+    # data module configs
+    recipe.data.tokenizer = hf_tokenizer(HF_MODEL_URI)
+
+    # Compute dtype configs
+    if args.compute_dtype.lower() == "fp8":
+        recipe.trainer.plugins = bf16_with_fp8_mixed()
+        recipe.trainer.plugins.grad_reduce_in_fp32 = False
+
+    recipe.model.config.cross_entropy_fusion_impl = "te"
+    recipe.model.config.cross_entropy_loss_fusion = True
+    recipe.model.config.apply_rope_fusion = True
+    recipe.model.config.moe_permute_fusion = True
+    return recipe
+
+
+if __name__ == "__main__":
+    args = parse_cli_args().parse_args()
+    args_sanity_check(args)
+
+    kwargs = get_user_configs(args.gpu.lower(), "sft", "llama4", "e128", args)
+    (
+        num_nodes,
+        mbs,
+        gbs,
+        tp_size,
+        pp_size,
+        cp_size,
+        vp_size,
+        ep_size,
+        num_layers,
+        hidden_size,
+        etp_size,
+        enable_cuda_graphs,
+        use_mcore_fsdp,
+        recompute_layers,
+        activation_offload_layers,
+    ) = kwargs[0:15]
+
+    # Only configure recipe if finetuning will actually run
```

Comment:
what happens if you configure it anyway but just keep the  condition you have added:
`        if args.skip_finetuning is not True:
            exp.add(
                recipe,
                executor=executor,
                name=exp_name,
                plugins=plugins,
            )`